src/StateProvider.js

```diff
@@ -0,0 +1,146 @@
+import React, { createContext, useReducer } from 'react';
+
+export const StateContext = createContext();
+const initialState = {
```

Comment:
would it be better if we store this in the data.js?

Reply:
yes I will do this